src/main.rs

```diff
@@ -309,7 +357,13 @@ fn main() {
 						BuildRuleCommand::Command { command, .. } => command,
 					};
 					if opt.sleep_run {
-						std::thread::sleep(std::time::Duration::from_millis(2500 + i as u64 * 5123 % 2000));
+						let estimated_total_task_time = estimated_total_task_time_func(task);
+						let diff_ms: i64 = thread_rng().gen_range(-150, 150);
+						let actual_job_time = match estimated_total_task_time {
+							Some(t) => std::time::Duration::from_millis((t.as_millis() as i64 + diff_ms).max(0) as u64),
```

Comment:
Maybe it's easier to just keep all the durations in milliseconds as i64. Now you're converting back and forth a lot.

Also, it seems like [`as_millis`](https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_millis) isn't stable yet. (Should error though. Doesn't it?)